src/modules/languages/c.nix

```diff
@@ -2,20 +2,42 @@
 
 let
   cfg = config.languages.c;
+
+  # Default to GCC if no compiler is explicitly enabled
+  useGcc = cfg.gcc.enable || !cfg.clang.enable;
+  useClang = cfg.clang.enable;
 in
 {
   options.languages.c = {
     enable = lib.mkEnableOption "tools for C development";
+
+    clang.enable = lib.mkEnableOption "clang";
+    gcc.enable = lib.mkEnableOption "GCC";
   };
 
   config = lib.mkIf cfg.enable {
-    packages = with pkgs; [
-      stdenv
+    languages.c = {
+      clang.enable = lib.mkDefault false;
+      gcc.enable = lib.mkDefault false;
+    };
+
+    packages = (with pkgs; [
       gnumake
-      clang
-      ccls
-      gcc
       pkg-config
-    ];
+
+      ccls # LSP server
```

Comment:
I lately mostly used clangd. What's the advantage of ccls these days?
Also we have wrapper in nixpkgs that only work if you set certain environment variables such as NIX_CFLAGS_COMPILE and NIX_LDFLAGS. Are those set in devenv?